nrcan_etl/src/energuide/embedded/region.py

```diff
@@ -20,22 +57,68 @@ class Region(enum.Enum):
 
     @classmethod
     def _from_name(cls, name: str) -> typing.Optional['Region']:
-        snake_name = name.upper().replace(' ', '_')
-        return Region[snake_name] if snake_name in Region.__members__ else None
+        english_region = Region[name] if name in Region.__members__ else None
+        if english_region:
+            return english_region
+
+        if name in _FRENCH_NAMES:
+            return Region[_FRENCH_NAMES[name]]
+
+        if name in _ALTERNATIVE_NAMES:
+            return Region[_ALTERNATIVE_NAMES[name]]
+
+        return None
+
 
     @classmethod
     def _from_code(cls, code: str) -> typing.Optional['Region']:
-        code = code.upper()
+        code = code.replace('.', '')
         for region in Region:
-            if code == region.value:
+            if code == region.value or _ALTERNATIVE_CODES.get(code) == region.value:
                 return region
+
+        return None
+
+
+    @classmethod
+    def _from_fuzzy_match(cls, data: str) -> typing.Optional['Region']:
+        # Name is prefix of full name
+        for region in Region.__members__:
+            if region.startswith(data):
+                LOGGER.warning(f'PREFIX: Interpreting {data} as {region}')
+                return Region[region]
+
+        # Code separated by _
+        code = data.replace('_', '')
+        region = cls._from_code(code)
+        if region:
+            return region
+
+        # Levenshtein minimum number of edits to closest region
+        ratios = [(Levenshtein.ratio(region, data), region) for region in Region.__members__.keys() - {'UNKNOWN'}]
+        ratio, region = max(ratios)
+
+        if ratio > _MINIMUM_SIMILARITY:
+            LOGGER.warning(f'SIMILARITY: Interpreting {data} as {region}')
+            return Region[region]
+
         return None
 
     @classmethod
     def from_data(cls, data: str) -> 'Region':
-        output = cls._from_name(data)
+        if len(data) < 2:
+            return Region.UNKNOWN
```

Comment:
Wouldn't this case fall through and get `Region.UNKNOWN` on line 122?

Reply:
This is a guard against the prefix matching in the fuzzy matching. I don't want a single "N" to be a valid prefix match, since it's ambiguous (Could be newfoundland, nova scotia, northwest territories)